patches/json-c/0001-Add-managarm-specific-changes.patch

```diff
@@ -0,0 +1,25 @@
+From 79e757a57444fabd3647fc4a187345ab780502c0 Mon Sep 17 00:00:00 2001
+From: no92 <no92.mail@gmail.com>
+Date: Wed, 15 Jun 2022 13:08:10 +0200
+Subject: [PATCH] Add managarm-specific changes
+
+---
+ json_tokener.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/json_tokener.c b/json_tokener.c
+index 0c09b66..d74153a 100644
+--- a/json_tokener.c
++++ b/json_tokener.c
+@@ -327,7 +327,7 @@ struct json_object *json_tokener_parse_ex(struct json_tokener *tok, const char *
+ 		if (newloc == NULL)
+ 		{
+ 			freelocale(duploc);
+-			return NULL;
++			// return NULL;
```

Comment:
Why is this done?

Reply:
Due to our stubbing of locale functions, this prevents json-c from parsing input properly.

Reply:
Is the resulting code path correct? I'd expect that there is some reason that they return and don't ignore the error here. IMHO such cases should receive comments (or this should be fixed in mlibc).

Reply:
The parsing after this is correct. The problem that this is running into is that mlibc stubs locale functions, and returns nullptrs. This mostly works fine, but json-c treats this as an error condition and stops parsing, which is not what we want here.